### PR TITLE
[go_router_builder] Fix typo in README.md

### PR DESCRIPTION
**Pull request by riscait**
Changed `.go` to `.push`.
It is `.push` that returns Future, not `.go`.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

### CODE REVIEW

File: packages/go_router_builder/README.md [hunk 1]
```diff
@@ -384,4 +384,4 @@ An example is available [here](https://github.com/flutter/packages/blob/main/pac
 
 To run unit tests, run command `dart tool/run_tests.dart` from `packages/go_router_builder/`.
 
-To run tests in examples, run `flutter test` from `packages/go_router_builder/example`.
\ No newline at end of file
+To run tests in examples, run `flutter test` from `packages/go_router_builder/example`.
```
Comment (riscait) [inline packages/go_router_builder/README.md:387]: It appears that the formatter has added a new line at the end of the file.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
Thanks for the contribution!

The way to fix incorrect code samples is to adopt [code excerpts](https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#readme-code), which fixes the underlying problem rather than one specific symptom, and also ensures it doesn't regress as the package evolves.

**Comment by stuartmorgan** [comment]
If there are things that we want to show in the README but not the example app, they can be sourced from a readme_excepts.dart file made for that purpose ([example](https://github.com/flutter/packages/blob/main/packages/path_provider/path_provider/example/lib/readme_excerpts.dart)).

**Comment by chunhtai** [comment]
I am going to close this one since there is another pr that use code-excerpts https://github.com/flutter/packages/pull/7280

**Pull request closed without merging**